### PR TITLE
Fix endless loop in heka_tcp output plugin

### LINKED ISSUE #65: heka_tcp output plugin terminates if server is not listening anymore
Today I noticed that if I temporarily stop my heka_tcp input plugin, then the heka_tcp ouput plugin that were sending data to that stopped plugin will terminate quickly with the following error in logs:
```
1484245022748499356 [error] output.heka_tcp terminated: process_message() instruction_limit exceeded
```

Note that my heka_tcp input/output plugin are configured with ssl support.

### PR DESCRIPTION
**Pull request by bbinet**
Closes #65

### CODE REVIEW

**Comment by trink** [comment]
Please bump the version to 3.0.5 https://github.com/mozilla-services/lua_sandbox_extensions/blob/master/socket/CMakeLists.txt.socket#L6

**Comment by bbinet** [comment]
Done in d72308b

File: socket/sandboxes/heka/output/heka_tcp.lua [hunk 1]
```diff
@@ -59,26 +59,25 @@ end
...
-        if err == "timeout" or err == "closed" then
-            client:close()
-            client = nil
-            return -3, err
+local function send_message(msg)
+    local i = 1
+    for retry = 1, 3 do
+        local len, err, i = client:send(msg, i)
+        if len then
+            return 0
```
Comment (trink) [inline socket/sandboxes/heka/output/heka_tcp.lua]: This can lose data as it may have only sent part of the message

Comment (bbinet) [inline socket/sandboxes/heka/output/heka_tcp.lua]: Ok, I was thinking `client:send` would return `i>1` only when an error occurred.
I will rework my patch.

Comment (trink) [inline socket/sandboxes/heka/output/heka_tcp.lua]: http://w3.impa.br/~diego/software/luasocket/tcp.html#send but they are clearly not 100% complete/accurate

Comment (trink) [inline socket/sandboxes/heka/output/heka_tcp.lua]: @bbinet Will you hop on #hindsight

Comment (bbinet) [inline socket/sandboxes/heka/output/heka_tcp.lua]: on freenode?

**Comment by trink** [comment]
I am leaning towards dropping the retry within the plugin and just closing the connection on any error and having HS handle the retry.  I will test it out and it it looks good I will submit a PR for that.

**Comment by trink** [comment]
Closed by #68

**Pull request closed without merging**